### PR TITLE
[MM-60824] Replace FormattedMarkdownMessage in 'webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx' with FormattedMessage

### PR DESCRIPTION
**Pull request by TheInvincibleRalph**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28502

```release-note
NONE
```

### CODE REVIEW

**Comment by TheInvincibleRalph** [comment]
This is the new pull request for leaveTeamModal, @M-ZubairAhmed. Thank you for your review!

File: webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx [hunk 1]
```diff
@@ -83,66 +81,70 @@ export default class LeaveTeamModal extends React.PureComponent<Props, State> {
...
                     />
                 );
             } else if (numOfPublicChannels === 0) {
                 modalMessage = (
-                    <FormattedMarkdownMessage
-                        id='leave_team_modal_guest_only_private.desc'
-                        defaultMessage="** You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** You won't be able to rejoin it without an invitation from another team member. Are you sure?"
+                    <FormattedMessage
+                        id='leave_team_modal_guest_only_private.desc_updated'
+                        defaultMessage="You will be removed from <strong>{num_of_private_channels}</strong> private {num_of_private_channels,plural,one {channel} other {channels}} on this team. You won't be able to rejoin it without an invitation from another team member. Are you sure?"
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx:98]: as per original description, the strong tag is from `You will be ....on this team.` can you recheck all of the rest as well

File: webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx [hunk 2]
```diff
@@ -83,66 +81,70 @@ export default class LeaveTeamModal extends React.PureComponent<Props, State> {
...
                         }}
                     />
                 );
             } else if (numOfPublicChannels === 0) {
                 modalMessage = (
-                    <FormattedMarkdownMessage
-                        id='leave_team_modal_guest_only_private.desc'
-                        defaultMessage="** You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** You won't be able to rejoin it without an invitation from another team member. Are you sure?"
+                    <FormattedMessage
+                        id='leave_team_modal_guest_only_private.desc_updated'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx:97]:
```suggestion
                        id='leave_team_modal_guest_only_private.description'
```

File: webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx [hunk 3]
```diff
@@ -83,66 +81,70 @@ export default class LeaveTeamModal extends React.PureComponent<Props, State> {
...
                         }}
                     />
                 );
             } else {
                 modalMessage = (
-                    <FormattedMarkdownMessage
-                        id='leave_team_modal_guest_only_public.desc'
-                        defaultMessage="** You will be removed from {num_of_public_channels} public { num_of_public_channels,plural,one {channel} other {channels}} on this team.** You won't be able to rejoin it without an invitation from another team member. Are you sure?"
+                    <FormattedMessage
+                        id='leave_team_modal_guest_only_public.desc_updated'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx:108]:
```suggestion
                        id='leave_team_modal_guest_only_public.description'
```

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx:108]: similary for others as well

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Comment by TheInvincibleRalph** [comment]
Made the changes, @M-ZubairAhmed.

File: webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx [hunk 4]
```diff
@@ -83,66 +81,70 @@ export default class LeaveTeamModal extends React.PureComponent<Props, State> {
...
                     />
                 );
             } else if (numOfPublicChannels === 0) {
                 modalMessage = (
-                    <FormattedMarkdownMessage
-                        id='leave_team_modal_guest_only_private.desc'
-                        defaultMessage="** You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** You won't be able to rejoin it without an invitation from another team member. Are you sure?"
+                    <FormattedMessage
+                        id='leave_team_modal_guest_only_private.description'
+                        defaultMessage="<strong>You will be removed from {num_of_private_channels}</strong> private {num_of_private_channels,plural,one {channel} other {channels}} on this team. You won't be able to rejoin it without an invitation from another team member. Are you sure?"
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx:98]:
```suggestion
                        defaultMessage="<strong>You will be removed from {num_of_private_channels} private {num_of_private_channels,plural,one {channel} other {channels}} on this team.</strong> You won't be able to rejoin it without an invitation from another team member. Are you sure?"
```

I guess this is how it was previously, Can you check others as well

See previously
`** You will be removed from {num_of_private_channels} private { num_of_private_channels,plural,one {channel} other {channels}} on this team.** `

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx (hunk 1 above)
Comment (TheInvincibleRalph) [reply webapp/channels/src/components/leave_team_modal/leave_team_modal.tsx]: Noted.

**Comment by TheInvincibleRalph** [comment]
Updated, @M-ZubairAhmed. Thank you!

File: webapp/channels/src/i18n/en.json [hunk 5]
```diff
@@ -4061,12 +4061,12 @@
...
+  "leave_team_modal_private.description": "<strong>You will be removed from {num_of_private_channels}</strong> private {num_of_private_channels,plural,one {channel} other {channels}} on this team. If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?",
+  "leave_team_modal_public.description": "<strong>You will be removed from {num_of_public_channels}</strong> public {num_of_public_channels,plural,one {channel} other {channels}} on this team. Are you sure?",
+  "leave_team_modal.description": "<strong>You will be removed from {num_of_public_channels}</strong> public {num_of_public_channels,plural,one {channel} other {channels}} and <strong>{num_of_private_channels}</strong> private {num_of_private_channels,plural,one {channel} other {channels}} on this team. If the team is private you won't be able to rejoin it without an invitation from another team member. Are you sure?",
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/i18n/en.json:4069]: These didnt got updated with the changes you made can you cross verify

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
CI is failing can you check

File: webapp/channels/src/i18n/en.json (hunk 5 above)
Comment (TheInvincibleRalph) [reply webapp/channels/src/i18n/en.json:4069]: They got updated, I guess. The initial suffix was `desc` then I changed it to `description`

Comment (M-ZubairAhmed) [reply webapp/channels/src/i18n/en.json:4069]: Check the description

**Comment by TheInvincibleRalph** [comment]
> CI is failing can you check

I can't really say why the code is failing. I have checked the code and can't seem to find anything out of place. Is it something you can help with? Maybe I am missing something.

**Comment by TheInvincibleRalph** [comment]
The issue, I guess, is the inconsistent update in the `en.json` file like you rightly pointed out. I have fixed that and I am hoping the tests pass now. Thank you!

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by harshilsharma63 — APPROVED** [review]

**Comment by M-ZubairAhmed** [comment]
@TheInvincibleRalph can you update the test snapshots

**Comment by TheInvincibleRalph** [comment]
> @TheInvincibleRalph can you update the test snapshots

Roger

**Comment by TheInvincibleRalph** [comment]
Done, @M-ZubairAhmed

**Review by yasserfaraazkhan — APPROVED** [review]
image

**Pull request merged**